### PR TITLE
Update GCP auth docs for GCE identity token login

### PR DESCRIPTION
**Pull request by robmonte**
Adds to and updates gcp auth docs for feature added in 1.11 to allow easier login when running Vault on a GCE/GKE instance.

### CODE REVIEW

File: website/content/docs/auth/gcp.mdx [hunk 1]
```diff
@@ -346,22 +346,32 @@ Read more on the
 
 ### GCE
 
-GCE tokens **can only be generated from a GCE instance**. The JWT token can be
-obtained from the `service-accounts/default/identity` endpoint for a
-instance's metadata server.
+GCE tokens **can only be generated from a GCE instance**.
 
-#### curl Example
+1.  As of version 1.11, Vault can automatically discover the identity token on a GCE/GKE instance. This simplifies
```
Comment (taoism4504) [inline website/content/docs/auth/gcp.mdx:351]: Hi, please consider removing calling out version 1.11 within the text. You can just start the sentence with "Vault can automatically discover the identify ...".

Comment (robmonte) [reply website/content/docs/auth/gcp.mdx:351]: Makes sense, I'll get it changed. Thanks!

**Review by swenson — APPROVED** [review]
LGTM

**Review by austingebauer — APPROVED** [review]
👍

**Pull request merged**